### ISSUE TITLE
Setting AtomGroup positions results in wrong and inconsistent behavior

### ISSUE THREAD

**Issue by mnmelo** [MDAnalysis/mdanalysis#606]
AtomGroups currently cache a partial copy of the timestep (`_ts`), generated via `copy_slice`. This caching is only refreshed when timestep frames change.

However, no recaching is performed when `atomgroup.set_positions(newpos)` is invoked.
`set_positions` directly changes the trajectory timestep, but not the AtomGroup's cached copy.

This was puzzling, because all seems to work fine if `set_positions` happens to be invoked before `_ts` has been cached. However, any subsequent `set_positions` on the same AtomGroup, on the same frame, will fail as per above.

This may become moot with the AtomGroup overhaul (#363), but for the upcoming release it's a relatively serious bug.

Fix seems trivial. Will post a PR soon.

**Comment by richardjgowers** [comment]
Honestly, I don't understand the point of the local version `.ts` being cached?  Can we just remove the caching of this?

**Comment by mnmelo** [comment]
I can't comment on that. Not really sure what depends on it. On the other hand it's an obvious performance penalty, having to always copy coordinates.
I thought caching had to do with ensuring that the positions array is contiguous, but I now realize that the `get_positions` method goes directly to the Universe's `trajectory.ts` (with optional full copying of the array; neat).

Caching of the slice seems to simplify things  a little bit when writing out AtomGroups (in fact, I bumped into this problem when trying to write a trajectory). It is hard to search for because there are many instances of `object.ts` in the code that aren't specifically AtomGroup related.

I'm all for removing this apparently unneeded clutter. Can anyone else chip in?

**Comment by mnmelo** [comment]
Hmm, I went the quick and dirty way: removed the cached `.ts` and adapted the `Writer` code that uses it. Tests pass (except for `MDAnalysisTests.analysis.test_density.Test_density_from_Universe`, which is also failing on develop).

Shall I push these (potentially disruptive) changes?

**Comment by richardjgowers** [comment]
We could keep .ts but just make it uncached?  I thought it was only used in Writers, so should only be used once per frame anyway.

**Comment by mnmelo** [comment]
I think it's dangerous to have it uncached, since any access will trigger a (probably unneeded) deep copy of coordinates. And this side-effect is not evident from the unassuming name `.ts`.

I adapted the Writers to do their own coordinate copy when passed an array. The disadvantage now is that multiple writes of the same group, in the same frame, won't have the partial timestep cached.

Since nothing else seems to use `.ts` we could leave it cached, fix the problem I found above, but give it a more descriptive name? Perhaps even specifically make it a method, not a property? (like `get_partial_timestep()`)
